### PR TITLE
Improvements to service discovery tutorial

### PR DESCRIPTION
**Pull request by mattj-io**
## Description
https://jira.mesosphere.com/browse/DOCS-2038

## Urgency
- [ ] Blocker 
- [ ] High
- [ x] Medium

## Requirements
- Test all commands and procedures.
- Build content [locally](https://github.com/dcos/dcos-docs#test-local) and test for formatting/links.
- Add redirects to [dcos-website/redirect-files](https://github.com/dcos/dcos-website#managing-redirects).
- Change all affected versions (e.g. 1.7, 1.8, and 1.9).
- See the [contribution guidelines](https://github.com/dcos/dcos-docs#contributing).

### CODE REVIEW

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 1]
```diff
@@ -11,18 +11,31 @@ menu_order: 4
...
+
+`dcos node ssh --master-proxy --leader`
+
+# Mesos-DNS
+  
+  [Mesos-DNS](/docs/1.9/networking/mesos-dns/) assigns DNS entries for every Marathon app. The naming pattern for these entries is  *task.scheduler.mesos*
+  
+  The default scheduler for jobs is [Marathon](https://dcos.io/docs/1.10/overview/architecture/components/#marathon), so the Mesos-DNS name for your Redis service is *redis.marathon.mesos*.
+  
+  Let's use the [dig](https://linux.die.net/man/1/dig) command to retrieve the address record (also called the A record). Dig is a command line utility to query DNS servers. When used without argument it will use the system-wide configured DNS servers to query against, which in a DC/OS cluster is configured to point at Mesos-DNS:
```
Comment (sascala) [inline 1.10/tutorials/dcos-101/service-discovery.md]: comma after "argument"

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 2]
```diff
@@ -31,9 +44,14 @@ In this section, you will learn about DC/OS service discovery by exploring the d
   redis.marathon.mesos. 60  IN  A 10.0.0.43
   ```
 
-  The response tells us that the host for the `redis.marathon.mesos` is 10.0.0.43.
+  The response tells us that the host for the `redis.marathon.mesos` service is 10.0.0.43.
+
+  The A record only contains IP address information about the host. To connect to the service, you also need to know the port. In order to achieve this, Mesos-DNS also assigns an Service, or SRV, record for each Marathon app, which contains the port number.
```
Comment (sascala) [inline 1.10/tutorials/dcos-101/service-discovery.md]: "a Service"

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 3]
```diff
@@ -44,31 +62,34 @@ In this section, you will learn about DC/OS service discovery by exploring the d
   redis-1y1hj-s1.marathon.mesos. 60 IN  A 10.0.0.43
   ```
 
-  So you now know that your redis app is running on `10.0.0.43:30585`.
+  So you now know that your Redis app is running on `10.0.0.43:30585`
```
Comment (sascala) [inline 1.10/tutorials/dcos-101/service-discovery.md]: Remove "So you now know"

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 4]
```diff
@@ -44,31 +62,34 @@ In this section, you will learn about DC/OS service discovery by exploring the d
...
 
-  <a name="namedvips"></a>
-  * [Named Vips](/docs/1.10/networking/load-balancing-vips/) allow you to assign name/port pairs to your apps. Named VIPs allow you to assign meaningful names to your apps.
-  For example, you can assign a named VIP to your redis service by adding the following to the package definition:
+# Named Virtual IPs
+  
+  * [Named VIPs](/docs/1.10/networking/load-balancing-vips/) allow you to assign name/port pairs to your apps, which means you can give your apps meaningful names with a predictable port. They also provide in-built load balancing when using multiple instances of an application.
```
Comment (sascala) [inline 1.10/tutorials/dcos-101/service-discovery.md]: "built-in"

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 5]
```diff
@@ -44,31 +62,34 @@ In this section, you will learn about DC/OS service discovery by exploring the d
...
 
-  <a name="namedvips"></a>
-  * [Named Vips](/docs/1.10/networking/load-balancing-vips/) allow you to assign name/port pairs to your apps. Named VIPs allow you to assign meaningful names to your apps.
-  For example, you can assign a named VIP to your redis service by adding the following to the package definition:
+# Named Virtual IPs
+  
+  * [Named VIPs](/docs/1.10/networking/load-balancing-vips/) allow you to assign name/port pairs to your apps, which means you can give your apps meaningful names with a predictable port. They also provide in-built load balancing when using multiple instances of an application. 
+  For example, you can assign a named VIP to your Redis service by adding the following to the package definition:
 
   ~~~
```
Comment (sascala) [inline 1.10/tutorials/dcos-101/service-discovery.md]: Should ~~~ be ```?

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 6]
```diff
@@ -44,31 +62,34 @@ In this section, you will learn about DC/OS service discovery by exploring the d
...
 
   The full name is then generated using the following schema:
-  *vip-name.scheduler.l4lb.thisdcos.directory:vip-port*. In the example above, you can access the redis service from within the cluster at the following address: redis.marathon.l4lb.thisdcos.directory:6379.
+  *vip-name.scheduler.l4lb.thisdcos.directory:vip-port*. 
+
+  As we can see from the example [application](https://raw.githubusercontent.com/joerg84/dcos-101/master/app1/app1.py), this is the mechanism used by the redis package, and so you can access the Redis service from within the cluster at `redis.marathon.l4lb.thisdcos.directory:6379`.
 
 # Outcome
-You know how to use service discovery to connect to your application from within your DC/OS cluster.
+You know how to use service discovery to connect to your application from within your DC/OS cluster, and have learnt about the two mechanisms for service discovery available in DC/OS.
```
Comment (sascala) [inline 1.10/tutorials/dcos-101/service-discovery.md]: "learned"

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 7]
```diff
@@ -44,31 +62,34 @@ In this section, you will learn about DC/OS service discovery by exploring the d
...
 
 # Deep Dive
 What are the differences between [Mesos-DNS](#mesosdns) and [Named VIPs](#namedvips)?
 
 ## Mesos-DNS
-Mesos-DNS is a rather simple solution to finding applications inside the cluster. While DNS is supported by many applications, Mesos-DNS has the following drawbacks::
+Mesos-DNS is a simple DNS based solution to finding applications inside the cluster. While DNS is supported by many applications, Mesos-DNS has the following drawbacks:
```
Comment (sascala) [inline 1.10/tutorials/dcos-101/service-discovery.md]: DNS-based

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 8]
```diff
@@ -44,31 +62,34 @@ In this section, you will learn about DC/OS service discovery by exploring the d
...
 
-  * DNS caching: Applications sometimes cache DNS entries for efficiency and therefore might not have updated address information (e.g., after a task failure).
-  * You need to use SRV DNS records to retrieve information about the allocated ports. Even though applications commonly understand DNS A records, not all applications support SRV records.
+  * DNS caching: Applications sometimes cache DNS entries for efficiency and therefore might not have updated address information (e.g. after a task failure).
```
Comment (sascala) [inline 1.10/tutorials/dcos-101/service-discovery.md]: comma after "e.g."

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 9]
```diff
@@ -44,31 +62,34 @@ In this section, you will learn about DC/OS service discovery by exploring the d
...
+  * You need to use SRV DNS records to retrieve information about the allocated ports. Whilst applications commonly understand DNS A records, not all applications support SRV records.
```
Comment (sascala) [inline 1.10/tutorials/dcos-101/service-discovery.md]: "While applications..."

File: 1.10/tutorials/dcos-101/service-discovery.md (hunk 8 above)
Comment (mattj-io) [inline 1.10/tutorials/dcos-101/service-discovery.md]: I had to look this up - I did wonder initially why there were commas after all the e.g.'s and had no idea that this was a US English thing ;) https://jakubmarian.com/comma-after-i-e-and-e-g/ - is comma the standard ? If so I've taken out other ones ....

**Comment by mattj-io** [comment]
@sascala updates made as requested

**Comment by mattj-io** [comment]
Review request pls oss-team

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 10]
```diff
@@ -11,18 +11,31 @@ menu_order: 4
...
 
 # Objective
-Your application in the previous part used `redis.marathon.l4lb.thisdcos.directory:6379` as the address for connecting to redis. As redis might be running on any agent in the cluster (and furthermore on different ports), how does this address link to the actual running redis instance?
-In this section, you will learn about DC/OS service discovery by exploring the different options for service discovery for apps in DC/OS.
+Your [application](https://raw.githubusercontent.com/joerg84/dcos-101/master/app1/app1.py) in the previous part of this tutorial used `redis.marathon.l4lb.thisdcos.directory` as the address for connecting to Redis, with a port of 6379. As Redis might be running on any agent in the cluster, and potentially on different ports, how does this address resolve to the actual running Redis instance?
```
Comment (pleia2) [inline 1.10/tutorials/dcos-101/service-discovery.md]: Consistency, "app" vs "application"?

Comment (pleia2) [inline 1.10/tutorials/dcos-101/service-discovery.md]: You'll also need an extra newline after this if you want the sentence below to be its own paragraph.

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 11]
```diff
@@ -11,18 +11,31 @@ menu_order: 4
...
+In this section, you will learn about DC/OS service discovery by exploring the different options available for apps in DC/OS.
 
-# Steps
-  [Service discovery](/docs/1.10/networking/) allows you to connect to your applications without necessarily knowing where they are running. Service discovery is particularly useful in cases where applications may fail and be restarted on a different host.
+# Service Discovery
+  Service discovery allows you to connect to your applications without necessarily knowing where they are running. This is particularly useful in cases where applications may fail and be restarted on a different host.
```
Comment (pleia2) [inline 1.10/tutorials/dcos-101/service-discovery.md]: I liked the /networking link to the main docs about service discovery, why remove it?

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 12]
```diff
@@ -11,18 +11,31 @@ menu_order: 4
...
+  1. Named Virtual IPs.
+
+
+SSH into the Mesos master node in your cluster to see how these different service discovery methods work: 
+
+`dcos node ssh --master-proxy --leader`
+
+# Mesos-DNS
+  
+  [Mesos-DNS](/docs/1.9/networking/mesos-dns/) assigns DNS entries for every Marathon app. The naming pattern for these entries is  *task.scheduler.mesos*
```
Comment (pleia2) [inline 1.10/tutorials/dcos-101/service-discovery.md]: Update link to point to the 1.10 docs.

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 13]
```diff
@@ -44,31 +62,34 @@ In this section, you will learn about DC/OS service discovery by exploring the d
...
 # Outcome
-You know how to use service discovery to connect to your application from within your DC/OS cluster.
+You know how to use service discovery to connect to your application from within your DC/OS cluster, and have learned about the two mechanisms for service discovery available in DC/OS.
 
 # Deep Dive
 What are the differences between [Mesos-DNS](#mesosdns) and [Named VIPs](#namedvips)?
 
 ## Mesos-DNS
-Mesos-DNS is a rather simple solution to finding applications inside the cluster. While DNS is supported by many applications, Mesos-DNS has the following drawbacks::
+Mesos-DNS is a simple DNS-based solution to finding applications inside the cluster. While DNS is supported by many applications, Mesos-DNS has the following drawbacks:
```
Comment (pleia2) [inline 1.10/tutorials/dcos-101/service-discovery.md]: "DNS-based" seems a bit redundant since DNS is in the name :) I'd drop it.

**Comment by mattj-io** [comment]
@pleia2 - changes as requested

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 14]
```diff
@@ -11,18 +11,32 @@ menu_order: 4
...
 
 # Objective
-Your application in the previous part used `redis.marathon.l4lb.thisdcos.directory:6379` as the address for connecting to redis. As redis might be running on any agent in the cluster (and furthermore on different ports), how does this address link to the actual running redis instance?
-In this section, you will learn about DC/OS service discovery by exploring the different options for service discovery for apps in DC/OS.
+Your [app](https://raw.githubusercontent.com/joerg84/dcos-101/master/app1/app1.py) in the previous part of this tutorial used `redis.marathon.l4lb.thisdcos.directory` as the address for connecting to Redis, with a port of 6379. As Redis might be running on any agent in the cluster, and potentially on different ports, how does this address resolve to the actual running Redis instance?
```
Comment (joerg84) [inline 1.10/tutorials/dcos-101/service-discovery.md]: s/with a port of 6379/with port 6379.

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 15]
```diff
@@ -11,18 +11,32 @@ menu_order: 4
...
 
-# Steps
-  [Service discovery](/docs/1.10/networking/) allows you to connect to your applications without necessarily knowing where they are running. Service discovery is particularly useful in cases where applications may fail and be restarted on a different host.
+In this section, you will learn about DC/OS service discovery by exploring the different options available for apps in DC/OS.
```
Comment (joerg84) [inline 1.10/tutorials/dcos-101/service-discovery.md]: remove `for apps`. 

Why did you remove the sentence explaining service discovery?
Maybe we can reword to something like

```Service Discovery enables addressing of applications independent of where they are running in the cluster, which is particularly useful in cases where applications may fail and be restarted on a different host.```

Comment (joerg84) [reply 1.10/tutorials/dcos-101/service-discovery.md]: I see below, maybe the working is still useful...

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 16]
```diff
@@ -11,18 +11,32 @@ menu_order: 4
...
 
-  DC/OS provides two options for service discovery: Mesos-DNS and Named virtual IPs.
-  * SSH into your cluster to see how these service discovery methods work: `dcos node ssh --master-proxy --leader`,
-  <a name="mesosdns"></a>
-  * [Mesos-DNS](/docs/1.10/networking/mesos-dns/) assigns a Mesos-DNS for every Marathon app. The naming pattern is  *task.scheduler.mesos* and the default scheduler for jobs is `marathon`, so the Mesos-DNS name for your redis service is *redis.marathon.mesos*.
+# Service Discovery
+  [Service discovery](/docs/1.10/networking/) allows you to connect to your applications without necessarily knowing where they are running. This is particularly useful in cases where applications may fail and be restarted on a different host.
```
Comment (joerg84) [inline 1.10/tutorials/dcos-101/service-discovery.md]: Consider the proposal above.
It is not just you connecting to apps, it is mostly apps connecting to apps :).

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 17]
```diff
@@ -11,18 +11,32 @@ menu_order: 4
...
+  1. Named Virtual IPs.
+
+
+SSH into the Mesos master node in your cluster to see how these different service discovery methods work: 
+
+`dcos node ssh --master-proxy --leader`
+
+# Mesos-DNS
+  
+  [Mesos-DNS](/docs/1.10/networking/mesos-dns/) assigns DNS entries for every Marathon app. The naming pattern for these entries is  *task.scheduler.mesos*
```
Comment (joerg84) [inline 1.10/tutorials/dcos-101/service-discovery.md]: assign DNS entries–which are resolvable from any node in the cluster–  

s/for every Marathon app/for every task.

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 18]
```diff
@@ -31,9 +45,14 @@ In this section, you will learn about DC/OS service discovery by exploring the d
   redis.marathon.mesos. 60  IN  A 10.0.0.43
   ```
 
-  The response tells us that the host for the `redis.marathon.mesos` is 10.0.0.43.
+  The response tells us that the host for the `redis.marathon.mesos` service is 10.0.0.43.
```
Comment (joerg84) [inline 1.10/tutorials/dcos-101/service-discovery.md]: the response tells us that there is an instance of the `red.... ` service running on....

not 100% happy with my wording, but `the host` seems wrong as there could be multiple instances.

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 19]
```diff
@@ -31,9 +45,14 @@ In this section, you will learn about DC/OS service discovery by exploring the d
...
 
-  The A record only contains information about the host. To connect to the service, you also need to know the port. Use the following dig command to access the Service locator (SRV) DNS record, which also provides port information: `dig srv _redis._tcp.marathon.mesos`.
+  The A record only contains IP address information about the host. To connect to the service, you also need to know the port. In order to achieve this, Mesos-DNS also assigns a Service, or SRV, record for each Marathon app, which contains the port number.
```
Comment (joerg84) [inline 1.10/tutorials/dcos-101/service-discovery.md]: s/which contains/containing

the correspondence of which (app or record) seems more ambiguous otherwise

File: 1.10/tutorials/dcos-101/service-discovery.md [hunk 20]
```diff
@@ -44,31 +63,34 @@ In this section, you will learn about DC/OS service discovery by exploring the d
   redis-1y1hj-s1.marathon.mesos. 60 IN  A 10.0.0.43
   ```
 
-  So you now know that your redis app is running on `10.0.0.43:30585`.
+  This output tells you that your Redis app is running on `10.0.0.43:30585`
```
Comment (joerg84) [inline 1.10/tutorials/dcos-101/service-discovery.md]: service vs app (consistency, above you used service)

**Comment by mattj-io** [comment]
@joerg84 changes as requested
